### PR TITLE
Have signal SDK depend on only signal API.

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java [hunk 1]
```diff
@@ -66,8 +61,6 @@
 
   private static final String WORKER_THREAD_NAME =
       BatchSpanProcessor.class.getSimpleName() + "_WorkerThread";
-  private static final String SPAN_PROCESSOR_TYPE_LABEL = "spanProcessorType";
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java:69]: We'll need to decide whether we expose a `Listener` interface to be able to push these metrics via an extension. Presumably after this PR.

File: api/trace/src/main/java/io/opentelemetry/spi/trace/TracerProviderFactory.java [hunk 2]
```diff
@@ -5,7 +5,6 @@
 
 package io.opentelemetry.spi.trace;
```
Comment (anuraaga) [inline api/trace/src/main/java/io/opentelemetry/spi/trace/TracerProviderFactory.java:6]: I moved this file into the `api-trace` folder for now. Later we'd presumably move all the SPI into an `autoconfigure` or such artifact based on #2262

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
Do not remove metrics from processors yet.

**Comment by jkwatson** [comment]
I agree with Bogdan on this. I think we really do want those metrics in the processors and exporters; they are super valuable signals.

**Comment by bogdandrutu** [comment]
> I agree with Bogdan on this. I think we really do want those metrics in the processors and exporters; they are super valuable signals.

One caveat the dependency I think can be "implementation" because we do not expose any API from the metrics publicly in this artifact

**Comment by anuraaga** [comment]
I was thinking we'd need to expose a listener or such - how can we keep the metrics without forcing a user to use alpha APIs?

**Comment by bogdandrutu** [comment]
> I was thinking we'd need to expose a listener or such - how can we keep the metrics without forcing a user to use alpha APIs?

Let's discuss that in a different PR :))

Also you are going to invent a new metrics API "listener", we must go deeper :))

**Comment by anuraaga** [comment]
I was thinking that the next release seems like it will be quite breaking, so we would want to get as much breaking stuff in as possible. If we're splitting metrics out, ideally we split it everywhere by the next release so I made this PR. What "different PR" would be appropriate for discussiong? :P

**Comment by anuraaga** [comment]
Got one idea for a different PR let me play with it.

**Comment by anuraaga** [comment]
This looks like it got gobbled into #2341

**Pull request closed without merging**